Mathlib/Algebra/Order/SuccPred.lean

```diff
@@ -0,0 +1,189 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios, Yaël Dillies
+-/
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Algebra.Order.ZeroLEOne
+import Mathlib.Data.Int.Cast.Defs
+import Mathlib.Order.SuccPred.Basic
+
+/-!
+# Interaction between successors and arithmetic
+
+We define the `SuccAddOrder` and `PredSubOrder` typeclasses, for orders satisfying `succ x = x + 1`
+and `pred x = x - 1` respectively. This allows us to transfer the API for successors and
+predecessors into these common arithmetical forms.
+
+## Todo
+
+In the future, we will make `x + 1` and `x - 1` the `simp`-normal forms for `succ x` and `pred x`
+respectively. This will require a refactor of `Ordinal` first, as the `simp`-normal form is
+currently set the other way around.
+-/
+
+/-- A typeclass for `succ x = x + 1`. -/
+class SuccAddOrder (α : Type*) [Preorder α] [Add α] [One α] extends SuccOrder α where
+  succ_eq_add_one (x : α) : succ x = x + 1
+
+/-- A typeclass for `pred x = x - 1`. -/
+class PredSubOrder (α : Type*) [Preorder α] [Sub α] [One α] extends PredOrder α where
+  pred_eq_sub_one (x : α) : pred x = x - 1
+
+variable {α : Type*} {x y : α}
+
+namespace Order
+
+section Preorder
+
+variable [Preorder α]
+
+section Add
+
+variable [Add α] [One α] [SuccAddOrder α]
+
+theorem succ_eq_add_one (x : α) : succ x = x + 1 :=
+  SuccAddOrder.succ_eq_add_one x
```

Comment:
I would make this lemma simp so that people know not to use `succ` when `SuccAddOrder` is around
```suggestion
@[simp] theorem succ_eq_add_one (x : α) : succ x = x + 1 :=
  SuccAddOrder.succ_eq_add_one x
```

Reply:
Yeah, this would cause a bunch of issues with `Ordinal` as the simp normal form is currently set the other way around. That's a refactor I'll do after this PR gets merged.